src/gevent_server.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+"""Gevent implemnation of http server."""
+from server import response_error
+from server import response_ok
+from server import resolve_uri
+from server import parse_request
+
+
+def server(conn, address):
+    """Return message to client."""
+    try:
+        buffer_length = 8
+        reply_complete = False
+        full_string = u""
+        while not reply_complete:
+            part = conn.recv(buffer_length)
+            full_string = full_string + part.decode('utf-8')
+            if len(part) < buffer_length:
+                reply_complete = True
+        print(full_string)
+        try:
+            uri = parse_request(full_string)
+            body_tuple = resolve_uri(uri)
+            if body_tuple:
+                conn.send(response_ok(body_tuple))
+            else:
+                conn.sendall(response_error(u'404 Page Not Found'))
+            conn.close()
+        except NameError('Method not GET'):
+            conn.sendall(response_error(u'405 Method Not Allowed'))
+            conn.close()
+        except TypeError('HTTP protol incorrect'):
+            conn.sendall(response_error(u'505 HTTP Version Not Supported'))
+            conn.close()
+        except SyntaxError('URI incorrect'):
+            conn.sendall(response_error(u'404 Page Not Found'))
+            conn.close()
+        except IOError('File not found'):
+            conn.sendall(response_error(u'404 Page Not Found'))
+            conn.close()
```

Comment:
It would be preferable to simply build a response in each of these several `except` clauses.  It's quite repetitious to call `conn.sendall` in all these places. If you just built a response you could do something like this pseudocode:

```
try:
    uri = parse_request(request):
    body_tuple = resolve_uri(uri)
    if body_tuple:
        response = response_ok(body_tuple)
    else:
        response = response_error(u'404')
except OneType:
    response = response_error('400')
except AnotherType:
    response = response_error('500')
finally:
    conn.sendall(response)
    conn.close()
```